anesthetic/examples/perfect_ns.py

```diff
@@ -0,0 +1,197 @@
+"""Perfect nested sampling generators."""
+import numpy as np
+from anesthetic.examples.utils import random_ellipsoid
+from anesthetic import NestedSamples
+from anesthetic.samples import merge_nested_samples
+
+
+def gaussian(nlive, ndims, sigma=0.1, R=1, logLmin=-1e-2):
+    """Perfect nested sampling run for a spherical Gaussian & prior.
+
+    Up to normalisation this is identical to the example in John Skilling's
+    original paper https://doi.org/10.1214/06-BA127 . Both spherical Gaussian
+    width sigma and spherical uniform prior width R are centered on zero
+
+    Parameters
+    ----------
+    nlive: int
+        number of live points
+
+    ndims: int
+        dimensionality of gaussian
+
+    sigma: float
+        width of gaussian likelihood
+
+    R: float
+        radius of gaussian prior
+
+    logLmin: float
+        loglikelihood at which to terminate
+
+    Returns
+    -------
+    samples: NestedSamples
+        Nested sampling run
+    """
+
+    def logLike(x):
+        return -(x**2).sum(axis=-1)/2/sigma**2
+
+    def random_sphere(n):
+        return random_ellipsoid(np.zeros(ndims), np.eye(ndims), n)
+
+    samples = []
+    r = R
+    logL_birth = np.ones(nlive) * -np.inf
+    logL = logL_birth.copy()
+    while logL.min() < logLmin:
+        points = r * random_sphere(nlive)
+        logL = logLike(points)
+        samples.append(NestedSamples(points, logL=logL, logL_birth=logL_birth))
+        logL_birth = logL.copy()
+        r = (points**2).sum(axis=-1, keepdims=True)**0.5
```

Comment:
What shape does `points` have? Is it `(nlive, ndim)`? Such that we are taking the norm across `ndim` and end up with an array `r` of length `len(r)=nlive`?

Reply:
Yes.